core/good_key.go

```diff
@@ -63,9 +64,80 @@ func GoodKey(key crypto.PublicKey) error {
 // GoodKeyECDSA determines if an ECDSA pubkey meets our requirements
 func GoodKeyECDSA(key ecdsa.PublicKey) (err error) {
 	log := blog.GetAuditLogger()
-	err = NotSupportedError("ECDSA keys not yet supported")
-	log.Debug(err.Error())
-	return
+
+	// Check the curve.
+	err = GoodCurve(key.Curve)
+	if err != nil {
+		return err
+	}
+
+	// Assuming a prime field since a) we are only allowing such curves
+	// and b) crypto/elliptic only supports prime curves.
+	params := key.Params()
+
+	// Partial check of the public key for an invalid range in the EC group:
+	//   Verify that key is not the point at infinity O.
+	if isAdditiveIdentity(key.Curve, key.X, key.Y) {
+		err = MalformedRequestError("Key x, y must not be the point at infinity")
+		log.Debug(err.Error())
+		return err
+	}
+
+	// Verify that each coordinate of the public key has the unique correct representation of an
+	// element in the underlying field by verifying that x,y are integers in [0,P-1].
+	if key.X.Sign() < 0 || key.Y.Sign() < 0 {
+		err = MalformedRequestError("Key x, y must not be negative")
+		log.Debug(err.Error())
+		return err
+	}
+
+	if key.X.Cmp(params.P) >= 0 || key.Y.Cmp(params.P) >= 0 {
+		err = MalformedRequestError("Key x, y must not exceed P-1")
+		log.Debug(err.Error())
+		return err
+	}
+
+	// Ensure that public key is on the correct elliptic curve:
+	// verify that y**2 === x**3 - 3*x + B (mod P).
+	if !key.Curve.IsOnCurve(key.X, key.Y) {
+		err = MalformedRequestError("Key point is not on the curve")
+		log.Debug(err.Error())
+		return err
+	}
+
+	// Ensure that public key has the correct order:
+	// verify that n*Q = O.
+	ox, oy := key.Curve.ScalarMult(key.X, key.Y, params.N.Bytes())
+	if !isAdditiveIdentity(key.Curve, ox, oy) {
+		err = MalformedRequestError("Public key does not have correct order")
+		log.Debug(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func isAdditiveIdentity(c elliptic.Curve, x, y *big.Int) bool {
+	// The point at infinity (and additive identity) for all supported curves is
+	// (0,0). But we can implement this more generally by satisfying ourselves
+	// that the values provided are not the additive identity.
+	ax, ay := c.Add(x, y, x, y)
+	return ax.Cmp(x) == 0 && ay.Cmp(y) == 0
+}
+
+// GoodCurve determines if an elliptic curve meets our requirements.
+func GoodCurve(c elliptic.Curve) (err error) {
+	log := blog.GetAuditLogger()
+
+	// Simply use a whitelist for now.
+	switch c.Params().Name {
```

Comment:
The paranoid in me dislikes treating the Name field as authoritative here.  Since `CurveParams` is a struct, you can compare directly. This might look something like:

```
case "P-256" && params == elliptic.P256().Params():
    return nil
case "P-384" && params == elliptic.P384().Params():
   return nil
...
```